how to run ACCESS-OM3 beta release (#959)
ACCESS-NRI/access-hive.org.au: docs/models/run-a-model/run-access-om3.md

{% set model = "ACCESS-OM3" %}
{% set github_configs = "https://github.com/ACCESS-NRI/access-om3-configs" %}
{% set configs_docs = "https://access-om3-configs.access-hive.org.au" %}
{% set example_branch = "release-MC_25km_jra_ryf" %}
{% set example_folder = "25km_jra_ryf" %}
[cosima]: https://cosima.org.au
[PBS job]: https://opus.nci.org.au/display/Help/4.+PBS+Jobs
[payu]: https://github.com/payu-org/payu

> [model components]: /models/configurations/access-om/#model-components-{{{ model }}}

Reviewer: ```suggestion
[model components]: /models/configurations/access-om/#model-components-{{ model }}
```

[gadi]: https://opus.nci.org.au/display/Help/0.+Welcome+to+Gadi#id-0.WelcometoGadi-Overview

!!! release
    This is a **Beta Release**.<br>
    This model configuration and source code might change before the full release.<br>
    Limited support is currently provided for this model. Its usage is only recommended for testing by experienced users and collaborators. For a supported and validated model and configuration, see [Run ACCESS-OM2](/models/run-a-model/run-access-om2) instead.

<div class="text-card-group" markdown>
[:fontawesome-brands-github:{: class="twemoji icon-before-text"} {{ model }} configurations]({{github_configs}}){: class="text-card"}

> [![Hive](/assets/ACCESS_icon_HIVE.png){: class="icon-before-text"} {{ model }} configs docs]({{configs_docs}}){: class="text-card"}

Reviewer 2: I think the links of release notes would not change when you move them from the "draft" ACCESS-NRI topic to where they should finally reside, therefore you could start adding them I believe.

> [![Hive](/assets/ACCESS_icon_HIVE.png){: class="icon-before-text"} {{ model }} configs docs]({{configs_docs}}){: class="text-card"}

Reviewer: Just the link would be private until they are published ?

If you are unsure whether {{ model }} is the right choice for your experiment, take a look at the overview of [ACCESS Models](/models).

> All {{{ model }}} configurations are open source, licensed under [CC BY 4.0](https://creativecommons.org/licenses/by/4.0/?ref=chooser-v1")![CC icon](https://mirrors.creativecommons.org/presskit/icons/cc.svg?ref=chooser-v1){: style="height:1em;margin-left:0.2em;vertical-align:text-top;"}![BY icon](https://mirrors.creativecommons.org/presskit/icons/by.svg?ref=chooser-v1){: style="height:1em;margin-left:0.2em;vertical-align:text-top;"} and available on [ACCESS-NRI GitHub]({{github_configs}}).

Reviewer: ```suggestion
All {{model}} configurations are open source, licensed under [CC BY 4.0](https://creativecommons.org/licenses/by/4.0/?ref=chooser-v1")![CC icon](https://mirrors.creativecommons.org/presskit/icons/cc.svg?ref=chooser-v1){: style="height:1em;margin-left:0.2em;vertical-align:text-top;"}![BY icon](https://mirrors.creativecommons.org/presskit/icons/by.svg?ref=chooser-v1){: style="height:1em;margin-left:0.2em;vertical-align:text-top;"} and available on [ACCESS-NRI GitHub]({{github_configs}}).
```

### Get {{ model }} configuration

All released {{ model }} configurations are available from the [{{ model }} configs]({{ github_configs }}) GitHub repository; released configurations (`release-` in the branch name) are tested and supported by ACCESS-NRI. More information about the available experiments and the naming scheme of the branches can also be found in the [configuration documention](https://access-om3-configs.access-hive.org.au/), the [{{ model }} configs]({{ github_configs }}) GitHub repository and the ACCESS-HIVE Docs [{{model}}][model configurations] page.

> The first step is to choose a configuration from those available. For example, to run an ocean and sea ice configuration (MOM6-CICE6 at 25km horizontal resolution with repeat-year _JRA55_ forcing (without BGC), one should select the branch [`{{example_branch}}`](https://github.com/ACCESS-NRI/access-om3-configs/tree/{{example_branch}}).

Author: ```suggestion
The first step is to choose a configuration from those available. For example, to run an ocean and sea ice configuration (MOM6-CICE6) at 25km horizontal resolution with repeat-year _JRA55_ forcing (without BGC), one should select the branch [`{{example_branch}}`](https://github.com/ACCESS-NRI/access-om3-configs/tree/{{example_branch}}).
```

> 

Author: GitHub is [not letting me edit](https://github.com/orgs/community/discussions/4452) the following line:
>In the example above the `payu clone` command clones the 25km repeat-year JRA55 MOM6 (`M`) CICE6 (`C`) configuration (` -B {{ example_branch }}`) as a new experiment branch (`-b expt`) to a directory named `{{example_folder}}`.

I think this would be a good place to add our discussed comment about the importance of tags and releases.

Something like this could perhaps work:
>In the above command, the `-B {{ example_branch }}` chooses the {{ example_branch }} as a starting point. Note that this command will always take the tip of the branch. Currently, that is taking a tagged commit corresponding to the `1.0beta` [beta release](https://forum.access-hive.org.au/t/access-om3-25km-ryf-1-0-beta-release-note-draft/4703) however subsequent updates to this branch will mean that this command will not get the `1.0beta` release. If one wishes to specify a _tag_ on the branch then the following Payu command can be used... When giving feedback ([beta example](https://forum.access-hive.org.au/t/access-om3-25km-ryf-1-0-beta-feedback/4719)) or asking for help on the forum, it is helpful to specify the exact configuration version used.

Author: Sounds good :)

Author: Should probably add it to om2 too?

### Get {{ model }} configuration

> All released {{ model }} configurations are available from the [{{ model }} configs]({{ github_configs }}) GitHub repository; released configurations (`release-` in the branch name) are tested and supported by ACCESS-NRI. More information about the available experiments and the naming scheme of the branches can also be found in the [configuration documention](https://access-om3-configs.access-hive.org.au/), the [{{ model }} configs]({{ github_configs }}) GitHub repository and the ACCESS-HIVE Docs [{{model}}][model configurations] page.

Author: Yeah, as we've been discussing, I've been keen to move a lot of [that content from here](https://github.com/ACCESS-NRI/access-om3-configs/blob/main/README.md) into the config docs.

It's a later problem I think. How about we just remove the naming convention part and have configuration information (@minghangli-uni is about to add a bunch of material I think)

> All released {{ model }} configurations are available from the [{{ model }} configs]({{ github_configs }}) GitHub repository; released configurations (`release-` in the branch name) are tested and supported by ACCESS-NRI. More information about the available experiments and the naming scheme of the branches can also be found in the [configuration documention](https://access-om3-configs.access-hive.org.au/), the [{{ model }} configs]({{ github_configs }}) GitHub repository and the ACCESS-HIVE Docs [{{model}}][model configurations] page.

Reviewer: I think this is the only comment not addressed

> All released {{ model }} configurations are available from the [{{ model }} configs]({{ github_configs }}) GitHub repository; released configurations (`release-` in the branch name) are tested and supported by ACCESS-NRI. More information about the available experiments and the naming scheme of the branches can also be found in the [configuration documention](https://access-om3-configs.access-hive.org.au/), the [{{ model }} configs]({{ github_configs }}) GitHub repository and the ACCESS-HIVE Docs [{{model}}][model configurations] page.

Author: How about

> More information about the available experiments and the configurations can be found in the [configuration documention](https://access-om3-configs.access-hive.org.au/), the [{{ model }} configs]({{ github_configs }}) GitHub repository and the ACCESS-HIVE Docs [{{model}}][model configurations] page.

To find out more about configuration settings for the `config.yaml` file, refer to [how to configure your experiment with payu](https://payu.readthedocs.io/en/latest/config.html).

### Start from an existing restart

> By default, the configuration will start from a "cold-start", where initial conditions are set based on observations of salinity and temperature, but all other model variables are 0.

Author: I quite like this sentence, would be good to add to the om2 version too I think (GitHub isn't letting me do that as a suggestion.)

To extend or fork from an existing experiment, the model can be configured to start from an existing restart file.

To do this, add a `restart:` [entry](https://payu.readthedocs.io/en/latest/config.html#miscellaneous), with a path to the folder containing existing restart files, to the `config.yaml` file in the experiment.
Or, to do this automatically when setting up an experiment, add the `-r` flag to the `payu clone` command: 

> 

Author: I think it's more consistent here to add the actual command and then make it an animation? Something like this?

```suggestion
```bash
payu clone -b expt -B {{ example_branch }} -r ~/access-om3/prev_expt/archive/restart010 https://github.com/ACCESS-NRI/access-om3-configs.git {{example_folder}}
```
```

    <terminal-line>Created and checked out new branch: expt</terminal-line>
    <terminal-line>laboratory path:  /scratch/.../access-om3</terminal-line>
    <terminal-line>binary path:  /scratch/.../access-om3/bin</terminal-line>
    <terminal-line>input path:  /scratch/.../access-om3/input</terminal-line>
    <terminal-line>work path:  /scratch/.../access-om3/work</terminal-line>
    <terminal-line>archive path:  /scratch/.../access-om3/archive</terminal-line>
    <terminal-line>Updated metadata. Experiment UUID: daeee7ff-07e4-4f93-823b-cb7c6e4bdb6e</terminal-line>
    <terminal-line>Added 'restart: /scratch/.../access-om3/archive/prev_expt/restart010' to configuration file: config.yaml</terminal-line>
    <terminal-line>Added archive symlink to /scratch/.../access-om3/archive/{{example_folder}}-expt-daeee7ff</terminal-line>
    <terminal-line data="input" directory="~/access-om3">cd {{example_folder}}</terminal-line>
    <terminal-line data="input" directory="~/access-om3/{{example_folder}}"></terminal-line>
</terminal-window>

> 

Author: ```suggestion
!!! warning
    Note that the restart flag used here will only be applied if there is no restart directory in archive, and so does not have to be removed for subsequent submissions. See [Payu docs](https://payu.readthedocs.io/en/stable/config.html#miscellaneous) for further details.
    <br><br>
```

Author: As discussed [here](https://github.com/ACCESS-NRI/access-hive.org.au/issues/955#issuecomment-3086398484)

> ### Change run length

Author: In the OM2 case this section was before start from an existing restart, I would be in favor of them both being the same. To me, the run length is probably a more common thing than restart, so would suggest putting that first.

The run length is controlled  by`stop_option` and `stop_n`.<br>
Common options for `stop_option` are `ndays`, `nmonths` and `nyears`. `stop_n` provides the numerical count for `stop_option`.

The restart period is controlled by `restart_option` and `restart_n`, which set *how often* restart files are written.<br>
!!! tip

>     To be able to continue an experiment run, writing restart files at the end of an experiment run is required. To achieve this, the `restart_*` fields should be set as a divisor of the corresponding `stop_*` values. For long runs which are working well, as an additional post-processing step, it is possible to "prune" restart files using Payu, see: [Pruning model restarts](/models/run-a-model/run-access-om3/#pruning-model-restarts).

Reviewer: ```suggestion
    To be able to resume each run from a previous state, the model must save restart files at the end of each run. To achieve this, the `restart_*` fields should be set as a divisor of the time period defined by the `stop_*` values. For long and stable runs, disk usage can be reduced by pruning older restart files as a _payu_ post-processing step. For details, see: [Pruning model restarts](/models/run-a-model/run-access-om3/#pruning-model-restarts).
```
Per davide but with active voice

!!! warning
    MOM6 provides the ability to vertically remap diagnostics onto user-defined vertical coordinates, including density coordinates (check [MOM6 vertically remapped diagnostics documentation](https://mom6.readthedocs.io/en/main/api/generated/pages/Diagnostics.html#vertically-remapped-diagnostics) for more information). Remapping to density coordinates can add substantially to the runtime of the model. The default `diag_table` used by {{ model }} includes diagnostics remapped to density coordinates. These should be removed for performance reasons if they are not needed.

### Start the run from a specific restart file

By default, the configuration will start from a "cold-start", where initial conditions are set based on observations of salinity and temperature, but all other model variables are 0.

To extend or branch off from an existing experiment, the model can be configured to start from an existing restart file.

To do this, add a `restart:` [entry](https://payu.readthedocs.io/en/latest/config.html#miscellaneous), with a path to the folder containing existing restart files, to the `config.yaml` file in the experiment.

> Or, to do this automatically when setting up an experiment, add the `-r` flag to the `payu clone` command: 

Reviewer 2: I believe these options are the same for om2 and om3. 
Whatever option (`-r` or `--restart`) you think its best to point out, I would do the same for both pages.

> Or, to do this automatically when setting up an experiment, add the `-r` flag to the `payu clone` command: 

Reviewer: I think i made these two om2 and om3 sections consistent now

!!! warning
    MOM6 provides the ability to vertically remap diagnostics onto user-defined vertical coordinates, including density coordinates (check [MOM6 vertically remapped diagnostics documentation](https://mom6.readthedocs.io/en/main/api/generated/pages/Diagnostics.html#vertically-remapped-diagnostics) for more information). Remapping to density coordinates can add substantially to the runtime of the model. The default `diag_table` used by {{ model }} includes diagnostics remapped to density coordinates. These should be removed for performance reasons if they are not needed.

### Start the run from a specific restart file {: id='specific-restart'}

> By default, the configuration will start from a "cold-start", where initial conditions are set based on observations of salinity and temperature, but all other model variables are 0.

Reviewer: ```suggestion
By default, the configuration will start from a "cold-start", where initial conditions are set based on observations of salinity and temperature, but all other model variables are zero.
```

> !!! warning

Reviewer 3: Edit: in the above lines, both `yaml` and `YAML` are used. Be conisitent!

project: ol01
```

To use a `/scratch` storage allocation other than `project` (or your default if `project` is not set) then also set `shortpath` (e.g. to the desired `/scratch/PROJECT_CODE`). 

!!! warning

>     If changing projects during an experiment, set the `shortpath` field so that it's consistent for all runs of an experiment.<br>

Reviewer: ```suggestion
    If changing projects during an experiment, set the `shortpath` field so that it's consistent for all runs of an experiment.
```

```
To enable syncing, change `enable` to `True`, and set `path` to a location on `/g/data`, where _payu_ will copy output and restart folders. A sensible `path` could be: `/g/data/$PROJECT/$USER/{{model}}/experiment_name/`.

### Saving model restarts

By default, {{ model }} outputs restart files after every run to allow for subsequent runs to start from a previously saved model state.<br>
Restart files can occupy a significant amount of disk space, and keeping a lot of them is often not necessary.

> _Payu_ is configured to prune the number of restart files kept. By default, _payu_ will only keep most recent couple of restarts and the restarts which match the frequency set in the `restart_freq` field of `config.yaml`. This is commonly set to `1YS` to keep one restart per year, or `5YS` for a restart every fifth year. For more information and to change this setting, check [_payu_ Configuration Settings documentation](https://payu.readthedocs.io/en/latest/config.html#model).

Reviewer: Hmm clock_attributes control the model behavior and this section is about payu behavior. So the model might produce yearly restarts and then payu only keeps every 5th year. 

Not quite sure how to clarify that yet !

> _Payu_ is configured to prune the number of restart files kept. By default, _payu_ will only keep most recent couple of restarts and the restarts which match the frequency set in the `restart_freq` field of `config.yaml`. This is commonly set to `1YS` to keep one restart per year, or `5YS` for a restart every fifth year. For more information and to change this setting, check [_payu_ Configuration Settings documentation](https://payu.readthedocs.io/en/latest/config.html#model).

Reviewer: @chrisb13 - if you have a rewording of this that would be great

    enable: False # set path below and change to true
    path: none # Set to location on /g/data or a remote server and path (rsync syntax)
```
To enable syncing, change `enable` to `True`, and set `path` to a location on `/g/data`, where _payu_ will copy output and restart folders. A sensible `path` could be: `/g/data/$PROJECT/$USER/{{ model }}/experiment_name/`.

### Saving model restarts

> By default, {{ model }} outputs restart files after every run to allow for subsequent runs to start from a previously saved model state.<br>

Reviewer 2: I like @chrisb13's suggestion better, but I would still try to make it clearer that payu pruning only happens at the end of an entire experiment (which can be constituted of multiple continuous runs), while the restart period is a single run. 
Something like:
```suggestion
By default, {{ model }} saves restart files after each run, allowing subsequent simulations to resume from a previously saved model state. The default {{ model }} run length and restart period can be changed (see [Change run length and restart period](#change-run-length-and-restart-period)).<br>
However, restart files can occupy significant disk space, and keeping all of them at the end of a whole experiment is often not necessary. If disk space is limited, consider using _payu_'s restart files pruning feature. This allows you to keep all the restart files during an experiment (constituted by multiple runs) in case of a crash, while reducing disk usage after the whole experiment completes by automatically removing redundant restart files.
```

> By default, {{ model }} outputs restart files after every run to allow for subsequent runs to start from a previously saved model state.<br>

Reviewer: The pruning does happen every run, if there is something to prune

> By default, {{ model }} outputs restart files after every run to allow for subsequent runs to start from a previously saved model state.<br>

Reviewer 2: Ok, then how does `payu` pruning integrate with the restart writing period? 

Let's say I write restarts every 1 month (`restart_n`/`restart_period` = `1`/`nmonths`).
And then I set `restart_freq = 6M` (6 months I guess?).

Which restarts will I have at the end of every run (considering payu pruning)? 

Based on my understanding, I will keep all restarts until the 6th run, at the end of which the restarts will be pruned and only the latest 2 (5th run and 6th run) will be left. Then the restarts until the 12th run will all be kept until the end of this run, which prunes again, so that only the following restarts are present: 6th run, 11th run, 12th run. And so on.

Is that correct?

> By default, {{ model }} outputs restart files after every run to allow for subsequent runs to start from a previously saved model state.<br>

Reviewer 2: Ok, I would try to make it clearer then, because from the text I think it's not clear. 
Maybe an additional example could be helpful to simplify the explanation.
